torcheeg/models/cnn/fbcnet.py

```diff
@@ -207,4 +178,144 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
+                 num_classes: int = 4,
+                 stride_factor: int = 4, 
+                 temporal: str ='LogVarLayer',
+                 num_feature: int = 36, 
+                 dilatability: int = 8 ):
+
+        super(FBMSNet, self).__init__()
+        
+        self.in_channels = in_channels
+        self.num_electrodes = num_electrodes
+        self.chunk_size = chunk_size
+        self.stride_factor = stride_factor
+
+
+        try:
+            self.mixConv2d = nn.Sequential(
+                MixedConv2d(in_channels=in_channels, out_channels=num_feature, kernel_size=[(1,15),(1,31),(1,63),(1,125)],
+                            stride=1, padding='', dilation=1, depthwise=False),
+                nn.BatchNorm2d(num_feature),
+            )
+            self.scb = self.SCB(in_chan=num_feature, out_chan=num_feature*dilatability, num_electrodes=int(num_electrodes))
+
+            # Formulate the temporal agreegator
+            if temporal == 'VarLayer':
+                self.temporal_layer = VarLayer(dim=3)
+            elif temporal == 'StdLayer':
+                self.temporal_layer = StdLayer(dim=3)
+            elif temporal == 'LogVarLayer':
+                self.temporal_layer = LogVarLayer(dim=3)
+            elif temporal == 'MeanLayer':
+                self.temporal_layer = MeanLayer(dim=3)
+            elif temporal == 'MaxLayer':
+                self.temporal_layer = MaxLayer(dim=3)
+            else:
+                raise NotImplementedError
+            size = self.feature_dim(in_channels,num_electrodes, chunk_size)
+
+            self.fc = self.LastBlock(size[1],num_classes)
+        except:
+            raise Exception("Model init failed: The Chunksize must be a  multiple of stride_factor.Please modify values of stride_factor or chunk_size.")
+
+
+    def SCB(self, in_chan, out_chan, num_electrodes, weight_norm=True, *args, **kwargs):
+        return nn.Sequential(
+            Conv2dWithConstraint(in_chan, out_chan, (num_electrodes, 1), groups=in_chan,
+                                 max_norm=2, weight_norm=weight_norm, padding=0),
+            nn.BatchNorm2d(out_chan),
+            swish()
+        )
+    
+    def LastBlock(self, inF, outF, weight_norm=True, *args, **kwargs):
+        return nn.Sequential(
+            LinearWithConstraint(inF, outF, max_norm=0.5, weight_norm=weight_norm, *args, **kwargs),
+            nn.LogSoftmax(dim=1))
+
+    
+    def forward(self, x):
+        r'''
+        Args:
+            x (torch.Tensor): EEG signal representation, the ideal input shape is :obj:`[n, in_channel, num_electrodes, chunk_size ]`. Here, :obj:`n` corresponds to the batch size
+
+        Returns:
+            torch.Tensor[size of batch,number of classes],torch.Tensor[size of batch, length of deep feature code]: The first value is the predicted probability that the samples belong to the classes. The second value is the extracted deep features in the network.
+        '''
+        if len(x.shape) == 5:
```

Comment:
What is this used for? It's not expected for the user to input a 5-dimensional tensor.